git_greppers/processing.sh

```
@@ -0,0 +1,281 @@
+  
+  # Check if this is a merge commit (has more than one parent)
+  local parent_count=$(git cat-file -p "$commit_hash" | grep -c "^parent ")
+  
+  debug 2 "Commit $commit_hash has $parent_count parents"
+  
+  if [[ $parent_count -gt 1 ]]; then
+    is_merge=true
+    debug 2 "This is a merge commit"
+    
+    # Get parent commit hashes
+    local parent_hashes=($(git cat-file -p "$commit_hash" | grep "^parent " | sed 's/^parent //'))
```

Shellcheck (0.10.0)

[warning] Prefer mapfile or read -a to split command output (or quote to avoid splitting).

(SC2207)